config/configmapprovider/env.go

```diff
@@ -18,31 +18,39 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"net/url"
 	"os"
 
 	"gopkg.in/yaml.v2"
 
 	"go.opentelemetry.io/collector/config"
 )
 
+const envSchemeName = "env"
+
 type envMapProvider struct {
 	envName string
 }
 
 // NewEnv returns a new Provider that reads the configuration from the given environment variable.
-func NewEnv(envName string) Provider {
-	return &envMapProvider{
-		envName: envName,
-	}
+//
+// This Provider supports "env" scheme, and can be called with an url with or without the scheme name:
+// `NAME_OF_ENVIRONMENT_VARIABLE`
+// `env://NAME_OF_ENVIRONMENT_VARIABLE`
+func NewEnv() Provider {
+	return &envMapProvider{}
 }
 
-func (emp *envMapProvider) Retrieve(context.Context, WatcherFunc) (Retrieved, error) {
-	if emp.envName == "" {
-		return nil, errors.New("config environment not specified")
+func (emp *envMapProvider) Retrieve(_ context.Context, url *url.URL, _ WatcherFunc) (Retrieved, error) {
+	if url.Scheme != "" && url.Scheme != envSchemeName {
+		return nil, ErrSchemeIsNotSupported
 	}
 
-	content := os.Getenv(emp.envName)
+	if url.Host == "" {
+		return nil, errors.New("config environment not specified")
+	}
 
+	content := os.Getenv(url.Host)
```

Comment:
This feels wrong as the value is not used as a host identifier. I wonder whether it might make more sense to specify that the --config flag can take values that specify a source by using <source_id>:<opaque_data>. This would allow sources where specifying them as a URI makes sense to do so but doesn't force sources like this one into that mold.

Reply:
Indeed, was thinking about this option as well. I liked the URL because it has RFC and clear requirements, but your suggestion also works if we make "source_id" required to ensure that if ":" in the opaque_data we parse correctly. Or we put a requirement that ":" needs to be escaped, but I think requiring the "source_id:" in the Provider API is fine, and for the "config" flag I think it is fine to expect that no files have ":"?

Reply:
Changed, let me know about the `Provider` and `File` (which follows the URL definition, which means it is impossible to have "file:relative/path", only "relative/path" is accepted without the scheme "file:".

Reply:
> and for the "config" flag I think it is fine to expect that no files have ":"?

Unless we're supporting VMS, that's probably fine :)